src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java

```diff
@@ -52,139 +43,150 @@
 @CommandLineProgramProperties(
-        summary = "Apply tranche filtering based on a truth VCF of known common sites of variation and score from VCF INFO field",
+        summary = "Apply tranche filtering based on a truth VCF of known common sites of variation and a score from VCF INFO field",
         oneLineSummary = "Apply tranche filtering",
         programGroup = VariantEvaluationProgramGroup.class
 )
-@DocumentedFeature
-@ExperimentalFeature
-public class FilterVariantTranches extends CommandLineProgram {
-    @Argument(fullName = StandardArgumentDefinitions.VARIANT_LONG_NAME,
-            shortName = StandardArgumentDefinitions.VARIANT_SHORT_NAME,
-            doc = "Input VCF file")
-    private String inputVcf = null;
 
+public class FilterVariantTranches extends TwoPassVariantWalker {
     @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
             shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
             doc = "Output VCF file")
     private String outputVcf = null;
 
-    @Argument(fullName = "snp-truth-vcf", shortName = "snp-truth-vcf", doc = "Input file of known common SNP sites.")
-    private String snpTruthVcf = null;
+    @Argument(fullName="tranche",
+            shortName="t",
+            doc="The levels of truth sensitivity at which to slice the data. (in percents, i.e. 99.9 for 99.9 percent and 1.0 for 1 percent)",
+            optional=true)
+    private List<Double> tranches = new ArrayList<Double>(Arrays.asList(99.9, 99.0, 90.0));
 
-    @Argument(fullName = "indel-truth-vcf", shortName = "indel-truth-vcf", doc = "Input file of known common INDEL sites.")
-    private String indelTruthVcf = null;
+    @Argument(fullName="resource",
+            shortName = "resource",
+            doc="A list of validated VCFs with known sites of common variation",
+            optional=false)
+    private List<FeatureInput<VariantContext>> resources = new ArrayList<>();
 
     @Argument(fullName = "info-key", shortName = "info-key", doc = "The key must be in the INFO field of the input VCF.")
     private String infoKey = GATKVCFConstants.CNN_1D_KEY;
 
-    @Argument(fullName = "max-sites", shortName = "max-sites", doc = "Maximum number of truth VCF sites to check.")
-    private int maxSites = 1200;
+    private VariantContextWriter vcfWriter;
+    private List<Double> snpScores = new ArrayList<>();
+    private List<Double> snpCutoffs = new ArrayList<>();
+    private List<Double> indelScores = new ArrayList<>();
+    private List<Double> indelCutoffs = new ArrayList<>();
 
-    @Argument(fullName="tranche",
-            shortName="t",
-            doc="The levels of truth sensitivity at which to slice the data. (in percents, i.e. 99.9 for 99.9 percent and 1.0 for 1 percent)",
-            optional=true)
-    private List<Double> tranches = new ArrayList<Double>(Arrays.asList(99.9, 99.0, 90.0));
+    @Override
+    public void onTraversalStart() {
+        tranches.sort(Double::compareTo);
```

Comment:
This should verify that there is at least one tranche or throw, and that the tranche values make sense.

Reply:
added.